### ISSUE TITLE
Wrong type on the Secret used as dockerconfigjson

### ISSUE THREAD

**Issue by cotoman** [emberstack/kubernetes-reflector#397]
Hello! We're trying to use reflector as a way to copy our registry key secret across multiple namespaces. Now it is deployed to the target namespace as expected. We suspect that due to the fact that its type is `Opaque` and not `kubernetes.io/dockerconfigjson`, multiple pods are failing to pull our private images.

Is there an annotation that I've missed, which would cover this scenario?

```
apiVersion: v1
kind: Secret
metadata:
 name: key
 annotations:
   reflector.v1.k8s.emberstack.com/reflects: "source-namespace/key"
type: kubernetes.io/dockerconfigjson ->> can't be added directly in the target secret
```

Thanks

**Comment by winromulus** [comment]
Hi
The source secret has to match the type. There is no way to change the type since they are different schemas (and have validation).

**Comment by cotoman** [comment]
@winromulus The problem here is that the source secret is type of `type: kubernetes.io/dockerconfigjson`, but the mirrored secret is `Opaque`. And we want to keep mirror secret in full sync, even the type. Is there a way to achieve that?

**Comment by winromulus** [comment]
@cotoman can you tell me more about the setup? are you using auto-reflections (reflector creates the secret) or did you create the destination yourself and added annotations?

**Comment by cotoman** [comment]
We've managed to solved the issue. Will leave here the solution in case someone's facing the same problem. Since controller can't copy source secret's type, basically what needs to be done is to create destination (mirrored) secret with desired type. Now since we got the type set to `kubernetes.io/dockerconfigjson` Kubernetes will through an error saying that a value has to be provided. What can be done here, is to supply `data[.dockerconfigjson]` with a dummy value. Once we have the secret created (with correct type), will leave reflector controller to overwrite dummy value with the value found in the source secret.
